internal/service/resource_pool_fetcher.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package service
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/machinebox/graphql"
+	"github.com/openshift-kni/oran-o2ims/internal/data"
+	"github.com/openshift-kni/oran-o2ims/internal/k8s"
+	searchmodel "github.com/stolostron/search-v2-api/graph/model"
+)
+
+//go:generate mockgen -source=resource_pool_fetcher.go -package=service -destination=resource_pool_fetcher_mock.go
+type ResourcePoolFetcher interface {
+	FetchItems(ctx context.Context) (resources data.Stream, err error)
+}
+
+type resourcePoolFetcher struct {
+	logger       *slog.Logger
+	cloudID      string
+	backendURL   string
+	backendToken string
+	graphqlQuery string
+	graphqlVars  *searchmodel.SearchInput
+	resources    data.Stream
+}
+
+func NewResourcePoolFetcher(
+	logger *slog.Logger,
+	cloudID, backendURL, backendToken string,
+	graphqlQuery string, graphqlVars *searchmodel.SearchInput) ResourcePoolFetcher {
+	return &resourcePoolFetcher{
```

Comment:
Can we use the "builder" approach to create this object? Otherwise when this already long list of parameters grow it will be hard to understand.